geospaas_processing/cli/copy.py

```diff
@@ -0,0 +1,108 @@
+"""
+Download files that are selected from the database using input criteria.
+"""
+import argparse
+import json
+import ntpath
+import os
+from os import path
+import shutil
+
+import django
+import geospaas_processing.cli.download as download_cli
+from django.contrib.gis.geos import GEOSGeometry
+from geospaas.catalog.managers import LOCAL_FILE_SERVICE
+from geospaas.catalog.models import Dataset
+
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'geospaas_processing.settings')
+django.setup()
+
+
+def main():
+    """
+    Copy the files based on the addressed stored in database to the destination folder.
+    The destination folder as well as criteria for finding the desired files are input of this
+    function.
+    """
+    arg = cli_parse_args()
+    cumulative_query = json.loads(arg.query) if arg.query else {}
+    if arg.geometry:
+        cumulative_query['geographic_location__geometry__intersects'] = GEOSGeometry(arg.geometry)
+    designated_begin, designated_end = download_cli.find_designated_time(
+        arg.rel_time_flag, arg.begin, arg.end)
+    cumulative_query['time_coverage_start__gte'] = designated_begin
+    cumulative_query['time_coverage_end__lte'] = designated_end
+    datasets = Dataset.objects.filter(**cumulative_query)
+    for dataset in datasets:
+        try:
+            source_paths = dataset.dataseturi_set.filter(service=LOCAL_FILE_SERVICE)
+        except IndexError:
+            continue
+        for source_path in source_paths:
+            if path.isfile(source_path.uri):
+                if arg.link:
+                    try:
+                        os.symlink(dst=arg.destination_path.rstrip(os.path.sep)
+                                   + os.path.sep+ntpath.basename(source_path.uri),
```

Comment:
I use ntpath because some users (in future) may come from the windows world and always use windows! By using ntpath are unconcerned that the address separation character does not cause a problem for those kind of users.

Reply:
Using `os.path` will automatically choose the right implementation of the module (`posixpath` or `ntpath`).
You should not use `posixpath` or `ntpath` directly except in very specific cases. This is not one of them.

Reply:
I don't know that! thanks. Done in the next commit.